core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -240,23 +247,80 @@ export class ContributorAdminDashboardPageComponent implements OnInit {
     this.languageDropdownShown = !this.languageDropdownShown;
   }
 
-  toggleActivityDropdown(): void {
-    this.activityDropdownShown = !this.activityDropdownShown;
+  getDateThatIsDaysBeforeToday(numberOfDaysBeforeToday: number): Date {
+    const today = new Date();
+    return new Date(
+      today.getTime() - 24 * 60 * 60 * 1000 * numberOfDaysBeforeToday
+    );
+  }
+
+  isValidStartDate(selectedStartDate: Date): boolean {
+    const today = new Date();
+    return (
+      today.getTime() >= selectedStartDate.getTime() &&
+      selectedStartDate.getTime() >= this.endDateForUsersLastActivity.getTime()
+    );
+  }
+
+  isValidEndDate(selectedEndDate: Date): boolean {
+    const today = new Date();
+    return (
+      today.getTime() >= selectedEndDate.getTime() &&
+      selectedEndDate.getTime() <= this.startDateForUsersLastActivity.getTime()
+    );
+  }
+
+  getNumberOfDaysSinceStartDateForLastActivity(): number {
+    const oneDay = 24 * 60 * 60 * 1000;
```

Comment:
It seems that `24 * 60 * 60 * 1000` is used in several places. Can you define this at the top of the component?

Reply:
done